tests/cookbook/test_write.py

```diff
@@ -24,12 +28,92 @@ def test_parquet_write_with_partitioning(tmp_path):
     pd_df = df.write_parquet(tmp_path, partition_cols=["Borough"])
 
     read_back_pd_df = daft.read_parquet(tmp_path.as_posix() + "/**/*.parquet").to_pandas()
-    assert_df_equals(df.exclude("Borough").to_pandas(), read_back_pd_df)
+    assert_df_equals(df.to_pandas(), read_back_pd_df)
 
     assert len(pd_df) == 5
     assert len(pd_df._preview.preview_partition) == 5
 
 
+def test_empty_parquet_write_without_partitioning(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+    df = df.where(daft.lit(False))
+    output_files = df.write_parquet(tmp_path)
+    assert len(output_files) == 0
+    assert len(output_files._preview.preview_partition) == 0
+
+
+def test_empty_parquet_write_with_partitioning(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+    df = df.where(daft.lit(False))
+    output_files = df.write_parquet(tmp_path, partition_cols=["Borough"])
+    assert len(output_files) == 0
+    assert len(output_files._preview.preview_partition) == 0
+
+
+def test_parquet_write_with_partitioning_readback_values(tmp_path):
+    df = daft.read_csv(COOKBOOK_DATA_CSV)
+
+    output_files = df.write_parquet(tmp_path, partition_cols=["Borough"])
+    output_dict = output_files.to_pydict()
+    boroughs = {"QUEENS", "MANHATTAN", "STATEN ISLAND", "BROOKLYN", "BRONX"}
+    assert set(output_dict["Borough"]) == boroughs
```

Comment:
Should we also assert that the output paths and partition values are as expected as well? I suppose that the UUID part of the filename makes the former a bit difficult.

Reply:
Ah I see that you have some assertions on the paths below, nevermind!